src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IUnaryOperatorExpression.vb

```diff
@@ -0,0 +1,1760 @@
+Class A
+    Function Method() As System.UInt64
+        Dim i As System.UInt64 = Nothing
+        Return Not Method()'BIND:"Not Method()"
+    End Function
+End Class
+
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.UInt64)
+  IInvocationExpression ( Function A.Method() As System.UInt64) (OperationKind.InvocationExpression, Type: System.UInt64)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of UnaryExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+
+        <Fact()>
+        Public Sub Test_UnaryOperatorExpression_Method_Not_System_Boolean()
+            Dim source = <![CDATA[
+Class A
+    Function Method() As System.Boolean
+        Dim i As System.Boolean = Nothing
+        Return Not Method()'BIND:"Not Method()"
+    End Function
+End Class
+
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUnaryOperatorExpression (UnaryOperationKind.BooleanBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.Boolean)
+  IInvocationExpression ( Function A.Method() As System.Boolean) (OperationKind.InvocationExpression, Type: System.Boolean)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of UnaryExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+
+        <Fact()>
+        Public Sub Test_UnaryOperatorExpression_Method_Not_System_Object()
+            Dim source = <![CDATA[
+Class A
+    Function Method() As System.Object
+        Dim i As System.Object = Nothing
+        Return Not Method()'BIND:"Not Method()"
+    End Function
+End Class
+
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IUnaryOperatorExpression (UnaryOperationKind.ObjectNot) (OperationKind.UnaryOperatorExpression, Type: System.Object)
+  IInvocationExpression ( Function A.Method() As System.Object) (OperationKind.InvocationExpression, Type: System.Object)
+    Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of UnaryExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+
+        <Fact()>
+        Public Sub Test_UnaryOperatorExpression_Type_Plus_E()
+            Dim source = <![CDATA[
+Class A
+    Function Method() As E
+        Dim i As E = Nothing
+        Return +i'BIND:"+i"
+    End Function
+End Class
+Enum E
+A
+B
+End Enum
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: E)
```

Comment:
Probably a design issue, it looks like language doesn't say explicitly that +/- are supported for enums (sections **11.13.1 Unary Plus Operator** and **11.13.2 Unary Minus Operator**).

---
In reply to: [108308547](https://github.com/dotnet/roslyn/pull/18136#discussion_r108308547) [](ancestors = 108308547)